metricbeat/module/redis/keyspace/keyspace.go

```diff
@@ -42,43 +37,27 @@ func init() {
 
 // MetricSet for fetching Redis server information and statistics.
 type MetricSet struct {
-	mb.BaseMetricSet
-	pool *rd.Pool
+	*redis.MetricSet
 }
 
 // New creates new instance of MetricSet
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
-	// Unpack additional configuration options.
-	config := struct {
-		IdleTimeout time.Duration `config:"idle_timeout"`
-		Network     string        `config:"network"`
-		MaxConn     int           `config:"maxconn" validate:"min=1"`
-		Password    string        `config:"password"`
-	}{
-		Network:  "tcp",
-		MaxConn:  10,
-		Password: "",
-	}
-	err := base.Module().UnpackConfig(&config)
+	ms, err := redis.NewMetricSet(base)
 	if err != nil {
 		return nil, err
```

Comment:
I'd add stack error context here so that error message shows something like "[part 1] _Redis module could not be created because_ [part 2, more context from `NewMetricset` function] _there was an error unpacking config_". It is descriptive and shows the origin of the error. If the user has many modules and metricsets it helps him detect where's the config error